processor/src/main/java/org/mapstruct/ap/internal/util/MapperConfiguration.java

```diff
@@ -42,19 +43,29 @@
     private final MapperPrism mapperPrism;
     private final MapperConfigPrism mapperConfigPrism;
     private final DeclaredType config;
+    private final FormattingMessager messager;
 
-    public static MapperConfiguration getInstanceOn(Element e) {
-        return new MapperConfiguration( MapperPrism.getInstanceOn( e ) );
+    public static MapperConfiguration getInstanceOn(Element e, FormattingMessager messager) {
+        return new MapperConfiguration( MapperPrism.getInstanceOn( e ), messager );
     }
 
-    private MapperConfiguration(MapperPrism mapperPrism) {
+    private MapperConfiguration(MapperPrism mapperPrism, FormattingMessager messager) {
         this.mapperPrism = mapperPrism;
+        this.messager = messager;
 
         if ( mapperPrism.values.config() != null ) {
-            // TODO #737 Only a declared type makes sense here; Validate and raise graceful error;
-            // Also validate that @MapperConfig is present
-            this.config = (DeclaredType) mapperPrism.config();
-            this.mapperConfigPrism = MapperConfigPrism.getInstanceOn( config.asElement() );
+            if ( mapperPrism.config() instanceof DeclaredType ) {
```

Comment:
I think the config itself even needs to be an interface.. so you can check that as well